charts/consul/test/unit/cni-daemonset.bats

```diff
@@ -0,0 +1,254 @@
+#!/usr/bin/env bats
+
+load _helpers
+
+@test "cni/daemonset: disabled by default" {
+  cd `chart_dir`
+  assert_empty helm template \
+      -s templates/cni-daemonset.yaml  \
+      .
+}
+
+@test "cni/daemonset: enabled with connectInject.cni.enabled=true and connectInject.transparentProxy.defaultEnabled=true" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/cni-daemonset.yaml  \
+      --set 'connectInject.cni.enabled=true' \
+      --set 'connectInject.transparentProxy.defaultEnabled=true' \
+      . | tee /dev/stderr |
+      yq 'length > 0' | tee /dev/stderr)
+  [[ "${actual}" == *"true"* ]]
+}
+
+@test "cni/daemonset: disabled with connectInject.enabled=false and connectInject.transparentProxy.defaultEnabled=true" {
+  cd `chart_dir`
+  assert_empty helm template \
+      -s templates/cni-daemonset.yaml  \
+      .
+}
+
+@test "cni/daemonset: throws error when connectInject.enabled=true and connectInject.transparentProxy.defaultEnabled=false" {
+  cd `chart_dir`
+  run helm template \
+      -s templates/cni-daemonset.yaml  \
+      --set 'connectInject.cni.enabled=true' \
+      --set 'connectInject.transparentProxy.defaultEnabled=false' \
+      -s templates/cni-daemonset.yaml  \
+      .
+
+  [ "$status" -eq 1 ]
+  [[ "$output" =~ "connectInject.transparentProxy.defaultEnabled must be true if connectInject.cni.enabled is true" ]]
+}
+
+@test "cni/DaemonSet: image defaults to global.imageK8S" {
+  cd `chart_dir`
+  local actual=$(helm template \
+      -s templates/cni-daemonset.yaml  \
+      --set 'connectInject.cni.enabled=true' \
+      --set 'global.imageK8S=foo' \
```

Comment:
I wonder if something isn't plumbed correctly here because most of these tests do not have `tproxy.DefaultEnabled` set and if I understand correctly from the daemonset templating the daemonset should NOT be rendered at all without:
`{{- if (and .Values.connectInject.cni.enabled .Values.connectInject.transparentProxy.defaultEnabled) }}`
?

i.e. we need `      --set 'connectInject.transparentProxy.defaultEnabled=true' \`
for all of the below tests?

Reply:
Ah, okay that makes sense.
Usually we are explicit about the flags being passed, but I think this is fine!